osquery/events/eventpublisherplugin.cpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <chrono>
+#include <ctime>
+#include <exception>
+#include <mutex>
+#include <thread>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <osquery/config/config.h>
+#include <osquery/core/flags.h>
+#include <osquery/core/system.h>
+#include <osquery/database/database.h>
+#include <osquery/events/eventfactory.h>
+#include <osquery/events/eventpublisherplugin.h>
+#include <osquery/events/eventsubscriber.h>
+#include <osquery/logger/logger.h>
+#include <osquery/registry/registry_factory.h>
+#include <osquery/sql/dynamic_table_row.h>
```

Comment:
Are all of these includes needed? Can we minimize them to reduce linkage requirements later on (since our current cmake binds include paths availability to target).